networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/gossip/topics/validation/AttestationValidator.java

```diff
@@ -152,6 +154,18 @@ private boolean isBeforeMinimumBroadcastTime(
     return currentTimeMillis.compareTo(minimumBroadcastTimeMillis) < 0;
   }
 
+  private boolean isFromFarFuture(
+          final Attestation attestation, final UnsignedLong currentTimeMillis) {
+    final UnsignedLong attestationSlotTimeMillis =
+            secondsToMillis(recentChainData
+                    .getGenesisTime()
+                    .plus(attestation.getEarliestSlotForForkChoiceProcessing()
+                            .times(UnsignedLong.valueOf(SECONDS_PER_SLOT))
+                    ));
+    return attestationSlotTimeMillis.compareTo(currentTimeMillis
+            .plus(secondsToMillis(MAX_FUTURE_SLOT_ALLOWANCE.times(UnsignedLong.valueOf(SECONDS_PER_SLOT))))) > 0;
```

Comment:
nit: Suggest extracting the `currentTimeMillis.plus` bit to a variable - something like `discardAttestationsAfterMillis`.  Just makes the comparison clearer.

Reply:
Makes sense. Done.